nautobot/dcim/tests/test_models.py

```diff
@@ -747,6 +749,64 @@ def test_site_forbidden_for_non_root(self):
         self.assertIn("must not have an associated Site", str(cm.exception))
 
 
+class PlatformTestCase(TestCase):
+    def setUp(self):
+        self.standard_platform = Platform(name="Cisco IOS", slug="cisco-ios", network_driver="cisco_ios")
+        self.custom_platform = Platform(name="Private Platform", slug="private-platform", network_driver="secret_sauce")
+
+    def test_network_driver_netutils_defaults(self):
+        """Test that a network_driver setting derives related fields from netutils by default."""
+        self.assertEqual(self.standard_platform.network_driver_mappings["ansible"], "cisco.ios.ios")
+        self.assertEqual(self.standard_platform.network_driver_mappings["hier_config"], "ios")
+        self.assertEqual(self.standard_platform.network_driver_mappings["netmiko"], "cisco_ios")
+        self.assertEqual(self.standard_platform.network_driver_mappings["ntc_templates"], "cisco_ios")
+        self.assertEqual(self.standard_platform.network_driver_mappings["pyats"], "iosxe")
+        self.assertEqual(self.standard_platform.network_driver_mappings["pyntc"], "cisco_ios_ssh")
+        self.assertEqual(self.standard_platform.network_driver_mappings["scrapli"], "cisco_iosxe")
+
+    def test_network_driver_unknown(self):
+        """Test that properties are not set if the network_driver setting is not known by netutils."""
+        self.assertNotIn("ansible", self.custom_platform.network_driver_mappings)
+        self.assertNotIn("hier_config", self.custom_platform.network_driver_mappings)
+        self.assertNotIn("netmiko", self.custom_platform.network_driver_mappings)
+        self.assertNotIn("ntc_templates", self.custom_platform.network_driver_mappings)
+        self.assertNotIn("pyats", self.custom_platform.network_driver_mappings)
+        self.assertNotIn("pyntc", self.custom_platform.network_driver_mappings)
+        self.assertNotIn("scrapli", self.custom_platform.network_driver_mappings)
```

Comment:
I did expose all possible fields through the UI. As far as graphql, I don't know if it would matter if this field remains a `GenericScalar` since you can't filter a `GenericScalar`.

![UI](https://user-images.githubusercontent.com/75227981/256919596-302fb66f-43c9-4499-ad95-143a9195327d.png)